src/pages/index.js

```diff
@@ -23,59 +79,98 @@ const userInfo = new UserInfo({
 
 // PopupWithForm Instances
 const editProfilePopup = new PopupWithForm(profileEditModal, (data) => {
-  userInfo.setUserInfo({
-    name: data["title"], // Uses name="title" from HTML
-    description: data["description"], // Uses name="description" from HTML
-  });
-  editProfilePopup.close();
+  editProfilePopup.setButtonText("Saving...");
+  api
+    .updateUserInfo({ name: data["title"], about: data["description"] })
+    .then((userData) => {
+      userInfo.setUserInfo({
+        name: userData.name,
+        description: userData.about,
+      });
+      editProfilePopup.close();
+    })
+    .catch(console.error)
+    .finally(() => editProfilePopup.setButtonText("Save"));
 });
 editProfilePopup.setEventListeners();
 
-// Updated Add Card Popup ✅
 const addCardPopup = new PopupWithForm(addCardModal, (data) => {
-  console.log("Form Data (Add Card): ", data); // Debugging
-
-  const cardElement = createCard({
-    name: data["title"] || "Untitled", // Uses name="title" from HTML
-    link: data["image url"] || "https://via.placeholder.com/150", // Uses name="image url" from HTML
-  });
-  cardSection.addItem(cardElement);
-  addCardPopup.close();
-  addCardForm.reset(); // reset -- Thank you!!!
-  addCardValidator.toggleButtonState(); // this will disable the button --- Thank You!!!
+  addCardPopup.setButtonText("Saving...");
+  api
+    .addCard({ name: data["title"], link: data["image url"] })
+    .then((cardData) => {
+      const cardElement = createCard(cardData);
+      cardSection.addItem(cardElement);
+      addCardForm.reset();
+      addCardValidator.toggleButtonState();
+      addCardPopup.close();
+    })
+    .catch(console.error)
+    .finally(() => addCardPopup.setButtonText("Save"));
 });
 addCardPopup.setEventListeners();
 
+const editAvatarPopup = new PopupWithForm("#edit-avatar-modal", (data) => {
+  editAvatarPopup.setButtonText("Saving...");
+  api
+    .updateUserAvatar(data.avatar)
+    .then((userData) => {
+      profileImage.src = userData.avatar;
```

Comment:
`      profileImage.src = userData.avatar;
`

Only methods of the `UserInfo` class should set/get the profile data (including the **avatar**)